src/game/common/modules/updatemodule.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file
+ *
+ * @author xezon
+ *
+ * @brief
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "updatemodule.h"
+
+#include "gamelogic.h"
+#include "thing.h"
+
+UpdateModule::UpdateModule(Thing *thing, const ModuleData *module_data) :
+    BehaviorModule(thing, module_data),
+    m_updatePhase(0), // ???
+    m_indexInLogic(-1) // ???
+{
+}
+
+void UpdateModule::CRC_Snapshot(Xfer *xfer)
+{
+    BehaviorModule::CRC_Snapshot(xfer);
+}
+
+void UpdateModule::Xfer_Snapshot(Xfer *xfer)
+{
+    uint8_t version = 1u;
+    xfer->xferVersion(&version, 1u);
+    BehaviorModule::Xfer_Snapshot(xfer);
+
+    xfer->xferUnsignedInt(&m_updatePhase);
+
+    if (xfer->Get_Mode() == XFER_LOAD) {
+        m_updatePhase &= 0xFFFFFFFC;
+        m_updatePhase |= Get_Update_Phase();
+
+        if (m_updatePhase == 0) {
+            Encode_Frame(g_theGameLogic->Get_Frame());
+        }
+        m_indexInLogic = -1;
+    }
+}
+
+void UpdateModule::Load_Post_Process()
+{
+    BehaviorModule::Load_Post_Process();
+}
+
+UpdateModuleInterface *UpdateModule::Get_Update()
+{
+    return static_cast<UpdateModuleInterface *>(this);
+}
+
+SleepyUpdatePhase UpdateModule::Get_Update_Phase() const
+{
+    return SLEEPY_UPDATE_PHASE_2;
+}
+
+BitFlags<DISABLED_TYPE_COUNT> UpdateModule::Get_Disabled_Types_To_Process() const
+{
+    return BitFlags<DISABLED_TYPE_COUNT>();
```

Comment:
should return DISABLEDMASK_NONE, consult Jonwil if unclear how as he did this kind of thing in a PR before

Reply:
Added.